tests/appsec/test_event_tracking_v2.py

```diff
@@ -0,0 +1,398 @@
+    "Accept-Language": "en-GB, *;q=0.5",
+    "Content-Language": "en-GB",
+    "Content-Type": "application/json; charset=utf-8",
+    "Host": "127.0.0.1:1234",
+    "User-Agent": "Benign User Agent 1.0",
+    "X-Forwarded-For": "42.42.42.42, 43.43.43.43",
+    "X-Client-IP": "42.42.42.42, 43.43.43.43",
+    "X-Real-IP": "42.42.42.42, 43.43.43.43",
+    "X-Forwarded": "42.42.42.42, 43.43.43.43",
+    "X-Cluster-Client-IP": "42.42.42.42, 43.43.43.43",
+    "Forwarded-For": "42.42.42.42, 43.43.43.43",
+    "Forwarded": "42.42.42.42, 43.43.43.43",
+    "Via": "42.42.42.42, 43.43.43.43",
+    "True-Client-IP": "42.42.42.42, 43.43.43.43",
+}
+
+USER_ID_SAFE = "user_id_safe"
+USER_ID_IN_RULE = "user_id_unsafe"
+LOGIN_SAFE = "login_safe"
+LOGIN_IN_RULE = "login_unsafe"
+
+
+def validate_metric_type_and_version(event_type, version, metric):
+    return (
+        metric.get("type") == "count"
+        and f"event_type:{event_type}" in metric.get("tags", ())
+        and f"sdk_version:{version}" in metric.get("tags", ())
+    )
+
+
+def validate_tags_and_metadata(span, prefix, expected_tags, metadata, unexpected_metadata):
+    if metadata is not None:
+        for key, value in metadata.items():
+            expected_tags[prefix + "." + key] = value
+
+    for tag, expected_value in expected_tags.items():
+        assert tag in span["meta"], f"Can't find {tag} in span's meta"
+        value = span["meta"][tag]
+        if value != expected_value:
+            raise Exception(f"{tag} value is '{value}', should be '{expected_value}'")
+
+    if unexpected_metadata is not None:
+        for key in unexpected_metadata:
+            tag = prefix + "." + key
+            assert tag not in span["meta"], f"Tag {tag} found in span's meta"
+
+    return True
+
+
+@features.event_tracking_sdk_v2
+@scenarios.appsec_ato_sdk
+class Test_UserLoginSuccessEventV2_Tags:
+    """Test tags created in AppSec User Login Success Event SDK v2"""
+
+    def get_user_login_success_tags_validator(self, login, user_id, metadata=None, unexpected_metadata=None):
+        def validate(span):
+            expected_tags = {
+                "appsec.events.users.login.success.usr.login": login,
+                "appsec.events.users.login.success.usr.id": user_id,
+                "usr.id": user_id,
+                "appsec.events.users.login.success.track": "true",
+                "_dd.appsec.events.users.login.success.sdk": "true",
+                "http.client_ip": "1.2.3.4",
+                "_dd.appsec.user.collection_mode": "sdk",
+            }
+
+            return validate_tags_and_metadata(
+                span, "appsec.events.users.login.success", expected_tags, metadata, unexpected_metadata
+            )
+
+        return validate
+
+    def setup_user_login_success_event(self):
+        headers = {
+            "X-Forwarded-For": "1.2.3.4",
+        }
+
+        metadata = {"metadata0": "value0", "metadata_number": 123, "metadata_boolean": True}
```

Comment:
in dotnet we force metadata to be only strings,, this wouldn't reach the app, is it possible to change for only strings, ie. "true", "123" ? If we allowed objects it would encourage boxing

Reply:
I will modify this to have 2 tests, one with only strings, and another with numbers etc, and in dotnet you can set it as `missing_feature`, because this test want to test that numbers/booleans are converted to strings.

Reply:
thank you!